components/browser/state/src/main/java/mozilla/components/browser/state/state/AppIntentState.kt

```diff
@@ -2,15 +2,17 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-package mozilla.components.browser.session.engine.request
+package mozilla.components.browser.state.state
 
 import android.content.Intent
 
-class LaunchIntentMetadata(
+/**
+ * State keeping track of app intents to launch.
+ *
+ * @param url the URL to launch in an external app.
+ * @param appIntent the [Intent] to launch.
+ */
+data class AppIntentState(
```

Comment:
We could also move this to concept-engine as `AppIntent` and change all our engine observer to pass along an instance of it, instead of a `url` and `intent`. Decided to add it here though to keep the diff smaller and because it doesn't make a difference?

Reply:
Yeah, looks good to me. If we ever get to reverse the dependency (`engine-*` uses `browser-state`) then having the model in `browser-state` makes sense too.